sql-plugin/src/main/301db/scala/com/nvidia/spark/rapids/shims/v2/GpuRegExpReplaceExec.scala

```diff
@@ -31,19 +31,12 @@ class GpuRegExpReplaceMeta(
 
   override def tagExprForGpu(): Unit = {
     expr.regexp match {
-      case Literal(null, _) =>
-        willNotWorkOnGpu(s"null pattern is not supported on GPU")
       case Literal(s: UTF8String, DataTypes.StringType) =>
-        val pattern = s.toString
-        if (pattern.isEmpty) {
-          willNotWorkOnGpu(s"empty pattern is not supported on GPU")
-        }
-
         if (GpuOverrides.isSupportedStringReplacePattern(expr.regexp)) {
           // use GpuStringReplace
         } else {
           try {
-            new CudfRegexTranspiler(replace = true).transpile(pattern)
+            new CudfRegexTranspiler(replace = true).transpile(s.toString)
```

Comment:
Seems like a waste to throw away the transpile and make us do it again later.  Curious why we don't cache the transpiled expression in the tagging and then pass it to the GPU exec node as a parameter when we convert to the GPU.

Reply:
Thanks .. I have re-implemented so that we only transpile once and cache the result in the meta classes.